### PR TITLE
opencoarrays: revision for gcc

### PR DESCRIPTION
**Pull request by fxcoudert**

### CODE REVIEW

**Comment by zbeekman** [comment]
Hi guys,
We are aware of this issue and will get a temporary work around released ASAP

**Comment by zbeekman** [comment]
Also, kudos on the rapid package updates!

**Comment by fxcoudert** [comment]
Another option is for us is not to perform systematic tests.

File: Formula/opencoarrays.rb [hunk 1]
```diff
@@ -13,7 +14,11 @@ class Opencoarrays < Formula
     sha256 "fa3e7e32db42b67198043aa0c22dcd6f814b234897d2cfd8c9195fe4fd1e017b" => :yosemite
   end
 
-  option "without-test", "Skip build time tests (not recommended)"
+  # As long as there are known failures with Homebrew's default compiler,
+  # we turn the tests off by default
+  # https://github.com/sourceryinstitute/OpenCoarrays/issues/374
+  # option "without-test", "Skip build time tests (not recommended)"
+  option "with-test", "Perform build time tests"
```
Comment (zbeekman) [inline Formula/opencoarrays.rb]: We should have a release out later today that "fixes" the build time tests with GCC 7. By fixes I mean that we've added a flag for OpenCoarrays developers so that by default tests for known regressions (`convert-before-put`) and partially implemented features (`async_comp_alloc`) are turned __*OFF*__ and are only enabled when `OPENCOARRAYS_DEVELOPER=ON` in the environment or the advanced CMake option `RUN_DEVELOPER_TESTS` is `ON`.

You can see the proposed changes at https://github.com/sourceryinstitute/OpenCoarrays/pull/375

**Review by zbeekman — COMMENTED** [review]
I would recommend against making this change, since we are going to be minting a new release ASAP with a flag that turns off tests with known failures upstream. Assuming publishing a new release will automatically trigger a formula update and a new bottled build, this formula revision should be unnecessary, and would need to be reverted as soon as the new release was live in homebrew.

See, for the proposed upstream fix: https://github.com/sourceryinstitute/OpenCoarrays/pull/375

**Comment by zbeekman** [comment]
> Another option is for us is not to perform systematic tests.

Why? This is an easy fix, and it is useful for us over at OpenCoarrays to know that users are building and using OpenCoarrays with some QA measures in place. If there are *new* and/or *unknown* bugs we want to find them and hear about them.

The release of GCC 7 just caught me out. We've been working hard to add new features (coarrays with allocatable components, experimental and only partially implemented right now) and to fix bugs. We had been hoping to resolve the regression and finish more of the implementation before the GCC 7 release, but I have known that an option to enable developer-only tests would be needed if GCC 7 was released before we could get around to those changes.

TL;DR; I should have made the proposed fix sooner, but we __*SHOULD KEEP*__ the tests on by default, if everyone is able to wait a little bit for me to mint a new release.

CC: @vehre @rouson @jerryd

**Review by zbeekman — CHANGES_REQUESTED** [review]
OpenCoarrays [1.8.10](https://github.com/sourceryinstitute/OpenCoarrays/releases/tag/1.8.10) has been released and the two failing tests have been silenced by default.

Because of this, I recommend closing this PR in favor of a version bump for the OpenCoarrays formula.

**Comment by fxcoudert** [comment]
Agreed and thanks for your fast action. Please open the PR for the new version.

**Comment by zbeekman** [comment]
Superseded by https://github.com/Homebrew/homebrew-core/pull/13254
